File: backend/api/serializers/notification.py
from rest_framework import serializers
from api.models import Notification
from .user import UserSerializer


class NotificationSerializer(serializers.ModelSerializer):
    actor = UserSerializer(read_only=True)

    class Meta:
        model = Notification
        fields = ['id', 'actor', 'notification_type', 'post', 'is_read', 'created_at']

File: frontend/src/types/notification.ts
import type { User } from './user';

export interface Notification {
  id: number;
  actor: User;
  notification_type: 'follow' | 'like' | 'comment' | 'story_reaction' | 'story_reply';
  post: number | null;
  is_read: boolean;
  created_at: string;
}
